contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1405 @@
+        uint256 _amount,
+        string calldata _beneficiary,
+        address[] calldata _listTCO2,
+        address _toucanPool
+    ) internal {
+        // The carbon to be retired should be transferred or swapped inside this contract before calling this function.
+        uint256 leftToBurn = _amount;
+
+        for (uint256 i = 0; i < _listTCO2.length && leftToBurn > 0; i++) {
+            // Start with the first address in the list and work your way down.
+
+            // Get the pools balance of TCO2
+            uint256 poolBalance = IERC20(_listTCO2[i]).balanceOf(_toucanPool);
+
+            // Error check for possible 0 balance / stale lists
+            if (poolBalance == 0) {} else {
+                address[] memory redeemERC20 = new address[](1);
+                redeemERC20[0] = _listTCO2[i];
+
+                uint256[] memory redeemAmount = new uint256[](1);
+
+                // Burn only pool balance if there are more pool tokens than available
+                if (leftToBurn > poolBalance) {
+                    // Redeem from pool
+                    redeemAmount[0] = poolBalance;
+                    IBaseCarbonTonne(_toucanPool).redeemMany(
+                        redeemERC20,
+                        redeemAmount
+                    );
+
+                    // Retire TCO2
+                    IToucanCarbonOffsets(_listTCO2[i]).retire(poolBalance);
+                    emit ToucanRetired(
+                        msg.sender,
+                        _beneficiary,
+                        _toucanPool,
+                        _listTCO2[i],
+                        poolBalance
+                    );
+
+                    leftToBurn -= poolBalance;
+                } else {
+                    // Redeem from pool
+                    redeemAmount[0] = leftToBurn;
+                    IBaseCarbonTonne(_toucanPool).redeemMany(
+                        redeemERC20,
+                        redeemAmount
+                    );
+
+                    // Retire TCO2
+                    IToucanCarbonOffsets(_listTCO2[i]).retire(leftToBurn);
+                    emit ToucanRetired(
+                        msg.sender,
+                        _beneficiary,
+                        _toucanPool,
+                        _listTCO2[i],
+                        leftToBurn
+                    );
+
+                    leftToBurn = 0;
+                }
+            }
+        }
+
+        require(
+            leftToBurn == 0,
+            "Not all pool tokens were burned. Please submit a larger TCO2 token list."
+        );
+    }
+
+    function _bondFees(address _toucanPool) internal returns (bool) {
+        // Bond cumulative fees to the DAO if bond is large enough
+        uint256 poolBalance = IERC20(_toucanPool).balanceOf(address(this));
+
+        uint256 bondPrice = IKlimaBondDepository(bondDepository[_toucanPool])
+            .bondPriceInUSD();
+
+        if ((poolBalance * (10**9)) / bondPrice > 10000000) {
```

Comment:
nit: can avoid the exponentiation here:
```suggestion
        if ((poolBalance * (1e9)) / bondPrice > 10000000) {
```

Reply:
Also updated the value being checked against to be 1e7, as this is really a check to make sure that the bonded amount would be greater than .01 KLIMA, and KLIMA has 9 decimals.